sw/device/silicon_creator/lib/BUILD

```diff
@@ -0,0 +1,431 @@
+    ],
+)
+
+cc_test(
+    name = "log_unittest",
+    srcs = ["log_unittest.cc"],
+    deps = [
+        ":error",
+        ":log",
+        "//sw/device/silicon_creator/lib/drivers:uart",
+        "//sw/device/silicon_creator/testing:mask_rom_test",
+        "@googletest//:gtest_main",
+    ],
+)
+
+cc_library(
+    name = "manifest_intf",
+    hdrs = [
+        "manifest.h",
+        "manifest_size.h",
+    ],
+    deps = [
+        ":epmp_intf",
+        ":error",
+        ":keymgr_binding",
+        ":sigverify_internal",
+        "//sw/device/lib/base",
+        "//sw/device/silicon_creator/lib/drivers:lifecycle",
+    ],
+)
+
+cc_library(
+    name = "manifest",
+    srcs = ["manifest.c"],
+    deps = [
+        ":manifest_intf",
+        "//hw/top_earlgrey/sw/autogen:top_earlgrey",
+    ],
+)
+
+cc_library(
+    name = "mock_manifest",
+    testonly = True,
+    hdrs = ["mock_manifest.h"],
+    deps = [
+        ":manifest_intf",
+        "//sw/device/silicon_creator/testing:mask_rom_test",
+        "@googletest//:gtest",
+    ],
+)
+
+cc_test(
+    name = "manifest_unittest",
+    srcs = [
+        "manifest_unittest.cc",
+        # sources included here because we need MOCK_CSR.
+    ],
+    defines = ["MOCK_CSR"],
+    deps = [
+        ":epmp_intf",
+        ":error",
+        ":keymgr_binding",
+        ":manifest_intf",
+        ":sigverify_internal",
+        "//hw/top_earlgrey/sw/autogen:top_earlgrey",
+        "//sw/device/lib/base",
+        "//sw/device/silicon_creator/lib/base:mock_csr",
+        "//sw/device/silicon_creator/lib/drivers:lifecycle",
+        "//sw/device/silicon_creator/testing:mask_rom_test",
+        "@googletest//:gtest_main",
+    ],
+)
+
+cc_library(
+    name = "otbn_util",
+    srcs = ["otbn_util.c"],
+    hdrs = ["otbn_util.h"],
+    deps = [
+        "//hw/top_earlgrey/sw/autogen:top_earlgrey",
+        "//sw/device/lib/base",
+        "//sw/device/silicon_creator/lib/base:abs_mmio",
+        "//sw/device/silicon_creator/lib/drivers:otbn",
+    ],
+)
+
+cc_library(
+    name = "shutdown",
+    srcs = ["shutdown.c"],
+    hdrs = ["shutdown.h"],
+    deps = [
+        ":error",
+        ":log",
+        "//hw/ip/keymgr/data:keymgr_regs",
+        "//hw/ip/lc_ctrl/data:lc_ctrl_regs",
+        "//hw/ip/otp_ctrl/data:otp_ctrl_regs",
+        "//hw/ip/sram_ctrl/data:sram_ctrl_regs",
+        "//hw/top_earlgrey/ip/flash_ctrl/data/autogen:flash_ctrl_regs",
+        "//hw/top_earlgrey/sw/autogen:top_earlgrey",
+        "//sw/device/lib/base",
+        "//sw/device/silicon_creator/lib/base:abs_mmio",
+        "//sw/device/silicon_creator/lib/drivers:alert",
+        "//sw/device/silicon_creator/lib/drivers:lifecycle",
+        "//sw/device/silicon_creator/lib/drivers:otp",
+    ],
+)
+
+cc_test(
+    name = "shutdown_unittest",
+    srcs = [
+        "shutdown.c",
+        "shutdown.h",
+        "shutdown_unittest.cc",
+    ],
+    defines = [
+        "OT_OFF_TARGET_TEST",
```

Comment:
This should definitely be set by the toolchain... perhaps something like having the transition define `OT_IS_DEVICE_TARGET` and condition on that not being defined.

Reply:
I don't know whether or not the transition will allow me to alter the default set of preprocessor symbols defined.  We certainly could use one of the symbols already defined by the compiler, such as `unix` or `linux` or `__x86_64__`.

For better or worse, supplying this define in the build (test) rule is isomorphic with the meson configuration.

Reply:
I think using a pre-existing define is a bit sketch... at best we should be hiding it behind a define in `macros.h`. I'll let you figure this out, but I want to make sure this is on our list.